### ISSUE TITLE
Crate structure question

### ISSUE THREAD

**Issue by ebkalderon** [amethyst/amethyst#42]
Currently, the crate dependency map of Amethyst looks like this:

```
+----------------------------------------------------+
|                      amethyst                      |
+----------------------------------------------------+
         ^                 ^             ^
         |                 |             |
+--------+--------+        |             |
| amethyst_engine |        |             |
+-----------------+        |             |
     ^      ^              |             |
     |      |              |             |
     |      +------------------------+   |
     |                     |         |   |
     |          +----------+---+ +---+---+-----------+
     +----------| amethyst_ecs | | amethyst_renderer |
                +--------------+ +-------------------+
```

Is there any reason to have the `amethyst_engine` crate at all? Can we not push its functionality upward to the `amethyst` crate? Doing so would simplify things quite a bit with no difference in functionality (see below).

```
+------------------------------------+
|              amethyst              |
+------------------------------------+
        ^                  ^
        |                  |
+-------+------+ +---------+---------+
| amethyst_ecs | | amethyst_renderer |
+--------------+ +-------------------+
```

**Comment by White-Oak** [comment]
Yes

**Comment by ebkalderon** [comment]
Resolved by #70.
